Sage/carbon
# fix(flat-table): ensure that pressing arrow and space keys scroll table when it is scrollable FE-6273

### Proposed behaviour


Ensures that arrow and space keys scroll `FlatTable` that has overflow set. The tab stop has been
moved to container further in the HTML tree to ensure that the scroll behaviour works as expected.

### Current behaviour


Scrollable tables are not scrolled when arrow or space key is pressed and table wrapper is focused.

### Checklist




#### QA


### Additional context



### Testing instructions


`flat-table--with-sticky-head-row-span-and-colspan` story should scroll the table when arrow or space key pressed
Confirm the other stories behave as expected (eg the expandable or clickable row stories still work)

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
